future-proof-iot/RIOT-rs
# coap: Early clean up steps

This is the first of several end-to-end PRs to come that each represent some progress on the CoAP road map (#226). At this early stage, this is not yet about moving things into the proper modules, but about refactoring and pulling the components into local modules.

* The fixed slots are replaced with an LRUish pool
* Writes to stdout are redirected to a configurable logger, which the main application directs into a buffer accessible through CoAP, more precisely, only through secure connections that presented a known credential, but not to those just connecting "without client authentication" (there is no such thing in EDHOC, but the client presenting a raw public key is the closest approximation thereof).
* The unsafety of turning a `&mut impl T` into the known implementation is enhanced from wild-west "I know what I'm doing" style to the unsafe-but-justifiable downcast of coap_message.
* Minimal authentication (boolean access to stdout) is considered
* Lakers is updated to provide access to C_I that was previously only available through knowledge of the peer's choices
* The terminal script is runnable standalone through published components on PyPI

## Testing

* Run with some USB example; I use `laze -C examples/coap build -b particle-xenon run`
* Run any number of terminals with `./fauxhoc.py`; notice that they all show the same log output (with up to a second of delay), but once a 5th is started, one of the others terminates.
* If you have lakers-python installed from git locally, you can also run `python3 fauxhoc.py --random-identity`. The client will use a random identity, but still verify the server. In return, the server will answer some of the more basic request, but refuse to send stdout.
